### PR TITLE
Fixes error message in `validate` method in `logical_types.py`

### PR DESCRIPTION
**Pull request by sbadithe**
The error message refers to an incompatible `dtype` when it should be `LogicalType`.

### CODE REVIEW

**Review by gsheni — DISMISSED** [review]

**Comment by gsheni** [comment]
Looks good but you'll have to update the string in some tests (as well as update the release notes)

File: woodwork/logical_types.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def validate(self, series, *args, **kwargs):
         valid_dtype = self._get_valid_dtype(type(series))
         if valid_dtype != str(series.dtype):
             raise TypeValidationError(
-                f"Series dtype '{series.dtype}' is incompatible with {self.type_string} dtype.",
+                f"Series dtype '{series.dtype}' is incompatible with {self.type_string} LogicalType.",
```
Comment (sbadithe) [inline woodwork/logical_types.py:86]: Should we include `should be {valid_dtype}` at the end of the message?

Comment (gsheni) [reply woodwork/logical_types.py:86]: Yep, we can

**Review by gsheni — COMMENTED** [review]
Will approve once valid_dtype is added

File: woodwork/logical_types.py (hunk 1 above)
Comment (sbadithe) [reply woodwork/logical_types.py:86]: Changed it to  `try converting to {valid_dtype}` to try to be consistent w/ the error msg in `column_accessor.py`. Not sure which is better

File: woodwork/tests/logical_types/test_logical_types.py [hunk 2]
```diff
@@ -271,7 +271,7 @@ def test_ordinal_validate(sample_series):
 
     new_type = "string"
     error_message = re.escape(
-        f"Series dtype '{new_type}' is incompatible with ordinal dtype.",
+        f"Series dtype '{new_type}' is incompatible with ordinal LogicalType.",
```
Comment (gsheni) [inline woodwork/tests/logical_types/test_logical_types.py:274]:
```suggestion
        f"Series dtype '{new_type}' is incompatible with ordinal LogicalType, try converting to {valid_dtype}",
```

**Review by gsheni — APPROVED** [review]

**Pull request merged**